lib/active_record_inherit_assoc.rb

```diff
@@ -1,21 +1,22 @@
 require 'active_record'
 
-case ActiveRecord::VERSION::MAJOR
-when 4
-  ActiveRecord::Associations::Builder::Association.valid_options << :inherit
-  ActiveRecord::Associations::Builder::Association.valid_options << :inherit_allowed_list
-when 5
-  # We can't add options into `valid_options` anymore.
-  # Here are the possible solutions:
-  #   * monkey patch Assocition::VALID_OPTIONS
-  #   * prepend the `valid_options` method
-  #   * create an Extension class and add it via ActiveRecord::Associations::Builder::Association.extensions
-  #
-  # I went with the first one out of simplicity.
-  ActiveRecord::Associations::Builder::Association::VALID_OPTIONS << :inherit
-  ActiveRecord::Associations::Builder::Association::VALID_OPTIONS << :inherit_allowed_list
+module ActiveRecordInheritBuildAssocPrepend
+  INHERIT_OPTIONS = %i[inherit inherit_allowed_list].freeze
+
+  case ActiveRecord::VERSION::MAJOR
+  when 4
+    def valid_options
+      super + INHERIT_OPTIONS
+    end
+  when 5, 6
```

Comment:
we'll need this patch for rails 7 too
```suggestion
  else
```